### PR TITLE
{bio,data}[foss/2019b] ITK v4.13.1, VMTK v1.4.0 w/ Python 3.7.4

### PR DESCRIPTION
**Pull request by deniskristak**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    'PATH': ['bin']
+}
+
+sanity_check_paths = {
+    'files': ['bin/vmtk', 'bin/vmtkactivetubes'],
+    'dirs': ['bin', 'lib', 'include'],
+}
+
+sanity_check_commands = [
+    "cd %(installdir)s/test_data && vmtkimagereader -ifile vase.vti "
```
Comment (boegel) [inline easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb:73]: Also check `vmtk --help`?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    ('zlib', '1.2.11'),
+    ('Mesa', '19.1.7'),
+    ('Tkinter', '3.7.4'),
+]
+
+preconfigopts = "export CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH:$EBROOTITK/lib/cmake/ITK-%s/ && " % local_itk_shortver
+
+local_sys_deps = ['DOUBLECONVERSION', 'EIGEN', 'EXPAT', 'HDF5', 'JPEG', 'PNG', 'TIFF', 'ZLIB']
+local_sys_cmake = ['-DITK_USE_SYSTEM_%s=ON' % d for d in local_sys_deps]
+configopts = ' '.join(local_sys_cmake)
```
[the remark this answers is not in the document]
Comment (deniskristak) [reply easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb:45]: done

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+separate_build_dir = True
+
+builddependencies = [('CMake', '3.15.3')]
+
+dependencies = [
+    ('Python', '3.7.4'),
+    ('VTK', '8.2.0', versionsuffix),
+    ('h5py', '2.10.0', versionsuffix),
+    ('SciPy-bundle', '2019.10', versionsuffix),
+    ('ITK', '4.13.1', versionsuffix),
```
[the remark this answers is not in the document]
Comment (deniskristak) [reply easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb:33]: done

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+local_vmtk_test_commit = "cbf09e4da03aa332c584b2709c4254017db9d4ef"
+postinstallcmds = [
+    'cp -r %(builddir)s/easybuild_obj/tests %(installdir)s',
+    "cd %(installdir)s && mkdir test_data && cd test_data && " +
+    "curl https://raw.githubusercontent.com/vmtk/vmtk-test-data/%s/input/vase.vti -O " % local_vmtk_test_commit
+]
+
+modextrapaths = {
+    'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages'],
+    'PATH': ['bin']
```
[the remark this answers is not in the document]
Comment (deniskristak) [reply easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb:64]: done

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb (hunk 1 above)
Comment (deniskristak) [reply easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb:73]: done

File: easybuild/easyconfigs/i/ITK/ITK-4.13.1-foss-2019b-Python-3.7.4.eb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+]
+
+dependencies = [
+    ('Python', '3.7.4'),
+    ('double-conversion', '3.1.4'),
+    ('expat', '2.2.7'),
+    ('HDF5', '1.10.5'),
+    ('libjpeg-turbo', '2.0.3'),
+    ('libpng', '1.6.37'),
+    ('LibTIFF', '4.1.0'),
```
[the remark this answers is not in the document]
Comment (deniskristak) [reply easybuild/easyconfigs/i/ITK/ITK-4.13.1-foss-2019b-Python-3.7.4.eb:37]: done

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+configopts += ' -DSUPERBUILD_INSTALL_PREFIX=%(installdir)s '
+configopts += ' -DVMTK_BUILD_TESTING=ON '
+
+preinstallopts = 'cd VMTK-Build && '
+
+local_vmtk_test_commit = "cbf09e4da03aa332c584b2709c4254017db9d4ef"
+postinstallcmds = [
+    'cp -r %(builddir)s/easybuild_obj/tests %(installdir)s',
+    "cd %(installdir)s && mkdir test_data && cd test_data && " +
+    "curl https://raw.githubusercontent.com/vmtk/vmtk-test-data/%s/input/vase.vti -O " % local_vmtk_test_commit
```
[the remark this answers is not in the document]
Comment (deniskristak) [reply easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb:59]: done

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    {'download_filename': '%s.tar.gz' % local_commit, 'filename': SOURCE_TAR_GZ},
+    {'filename': 'vase.vti', 'extract_cmd': 'cp %s %(builddir)s'},
+]
+checksums = [
+    '39aeb1de6bba8511dff45f0a69c9d6dc8d61f323739312ddceb22f8773e21459',  # VMTK-1.4.0.tar.gz
+    'b5f761d7a923ee19cd9cc8df7fbc794ffaa16aca101455fcdf9ceaa610ca3f57',  # vase.vti
+]
+
+
+separate_build_dir = True
```
Comment (boegel) [inline easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb:37]: This is already the default for `CMakeMake`, so drop this (and the duplicate empty line above)

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    'https://github.com/vmtk/vmtk/archive/',
+    "https://raw.githubusercontent.com/vmtk/vmtk-test-data/%s/input/" % local_vmtk_test_commit,
+]
+sources = [
+    {'download_filename': '%s.tar.gz' % local_commit, 'filename': SOURCE_TAR_GZ},
+    {'filename': 'vase.vti', 'extract_cmd': 'cp %s %(builddir)s'},
+]
+checksums = [
+    '39aeb1de6bba8511dff45f0a69c9d6dc8d61f323739312ddceb22f8773e21459',  # VMTK-1.4.0.tar.gz
+    'b5f761d7a923ee19cd9cc8df7fbc794ffaa16aca101455fcdf9ceaa610ca3f57',  # vase.vti
```
Comment (boegel) [inline easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb:33]: please change comment to `vase-20180313.vti`

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    "cp %(builddir)s/vase.vti %(installdir)s/"
+]
+
+modextrapaths = {
+    'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages'],
+}
+
+sanity_check_paths = {
+    'files': ['bin/vmtk', 'bin/vmtkactivetubes'],
+    'dirs': ['bin', 'lib', 'include'],
```
Comment (boegel) [inline easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb:79]: Maybe the `lib` check should be a bit more specific? `'lib/python%(pyshortver)s/site-packages'`?

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+sanity_check_commands = [
+    "cd %(installdir)s && vmtkimagereader -ifile vase.vti && rm vase.vti",
```
Comment (boegel) [inline easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb:83]: Hmm, copying the `vase.vti` to the install dir, and then removing it in the sanity check command is not very nice...

Can't we just run this command from `%(builddir)s` instead?

**Comment by deniskristak** [comment]
@boegel this PR should allow old ITK to pass through tests: 
https://github.com/easybuilders/easybuild-easyconfigs/pull/14046

File: easybuild/easyconfigs/i/ITK/ITK-4.13.1-foss-2019b-Python-3.7.4.eb [hunk 11]
```diff
@@ -0,0 +1,64 @@
+# Contributors:
+# Fenglai Liu (fenglai@accre.vanderbilt.edu) - Vanderbilt University
+# Alex Domingo (alex.domingo.toro@vub.be) - Vrije Universiteit Brussel (VUB)
+# Denis Kristak (INUITS)
+easyblock = 'CMakeMake'
```
[the remark this answers is not in the document]
Comment (deniskristak) [reply easybuild/easyconfigs/i/ITK/ITK-4.13.1-foss-2019b-Python-3.7.4.eb]: done

File: easybuild/easyconfigs/i/ITK/ITK-4.13.1-foss-2019b-Python-3.7.4.eb [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    ('zlib', '1.2.11'),
+    ('SWIG', '4.0.1'),
+]
+
+
+configopts = '-DCMAKE_BUILD_TYPE=Release -DITK_WRAP_PYTHON=ON '
+configopts += '-DBUILD_TESTING=OFF -DITK_USE_SYSTEM_SWIG=ON '
+configopts += '-DModule_ITKReview=ON -DModule_ITKVtkGlue=ON '
+
+preinstallopts = "export PYTHONPATH=%(installdir)s/lib/python%(pyshortver)s/site-packages:$PYTHONPATH && "
```
[the remark this answers is not in the document]
Comment (deniskristak) [reply easybuild/easyconfigs/i/ITK/ITK-4.13.1-foss-2019b-Python-3.7.4.eb]: done

File: easybuild/easyconfigs/i/ITK/ITK-4.13.1-foss-2019b-Python-3.7.4.eb [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+separate_build_dir = True
+
+modextrapaths = {'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages']}
```
[the remark this answers is not in the document]
Comment (deniskristak) [reply easybuild/easyconfigs/i/ITK/ITK-4.13.1-foss-2019b-Python-3.7.4.eb]: done

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+local_commit = '63a3b99'
+
+local_vmtk_test_commit = "cbf09e4da03aa332c584b2709c4254017db9d4ef"
+source_urls = [
+    'https://github.com/vmtk/vmtk/archive/',
+    "https://raw.githubusercontent.com/vmtk/vmtk-test-data/%s/input/" % local_vmtk_test_commit,
+]
+sources = [
+    {'download_filename': '%s.tar.gz' % local_commit, 'filename': SOURCE_TAR_GZ},
+    {'filename': 'vase.vti', 'extract_cmd': 'cp %s %(builddir)s'},
```
[the remark this answers is not in the document]
Comment (deniskristak) [reply easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb]: done

File: easybuild/easyconfigs/i/ITK/ITK-4.13.1-foss-2019b-Python-3.7.4.eb [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+    ('libpng', '1.6.37'),
+    ('LibTIFF', '4.0.10'),
+    ('VTK', '8.2.0', versionsuffix),
+    ('zlib', '1.2.11'),
+    ('SWIG', '4.0.1'),
+]
+
+
+configopts = '-DCMAKE_BUILD_TYPE=Release -DITK_WRAP_PYTHON=ON '
+configopts += '-DBUILD_TESTING=OFF -DITK_USE_SYSTEM_SWIG=ON '
```
[the remark this answers is not in the document]
Comment (deniskristak) [reply easybuild/easyconfigs/i/ITK/ITK-4.13.1-foss-2019b-Python-3.7.4.eb]: done

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb (hunk 7 above)
Comment (deniskristak) [reply easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb]: done

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb (hunk 8 above)
Comment (deniskristak) [reply easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb]: done

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb (hunk 9 above)
Comment (deniskristak) [reply easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb]: done

File: easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb (hunk 10 above)
Comment (deniskristak) [reply easybuild/easyconfigs/v/VMTK/VMTK-1.4.0-foss-2019b-Python-3.7.4.eb]: done

**Comment by deniskristak** [comment]
@boegel rdy for another review

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3160.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/5a0afdb5b1aeee27f06c6db0af6abbec for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (2 easyconfigs in total)
node3563.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/dc999539d3a4c7d388c6229e11a8654c for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @deniskristak!

**Pull request merged**